client/landing/gutenboarding/components/domain-picker/index.tsx

```diff
@@ -0,0 +1,67 @@
+/**
+ * External dependencies
+ */
+import React, { FunctionComponent, useState } from 'react';
+import { TextControl, Panel, PanelBody, PanelRow } from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
+import { __ as NO__ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { STORE_KEY as DOMAIN_STORE } from '../../stores/domain-suggestions';
+import { STORE_KEY as ONBOARD_STORE } from '../../stores/onboard';
+import { isFilledFormValue } from '../../stores/onboard/types';
+
+const DomainPicker: FunctionComponent = () => {
+	// User can search for a domain
+	const [ domainSearch, setDomainSearch ] = useState( '' );
+
+	// Without user search, we can provide recommendations based on title + vertical
+	const { siteTitle, siteVertical } = useSelect( select => select( ONBOARD_STORE ).getState() );
+
+	const suggestions = useSelect(
+		select => {
+			if ( domainSearch.trim() ) {
```

Comment:
Also [done in the selector](https://github.com/Automattic/wp-calypso/pull/37595/files#diff-9377c81b372a6aadbb0981b5c91c6524R77). Maybe drop one of them?

Reply:
Let's drop this on. All normalization should happen in the action creator 👍

Reply:
Wait, no. This is the state for the input and is a point where we decide to query based on other parameters (site title + vertical) or on the search query.

I think these are different concerns 🤔